aws/signer/internal/v4/headers_test.go

```diff
@@ -33,3 +33,31 @@ func TestAllowedQueryHoisting(t *testing.T) {
 		})
 	}
 }
+
+func TestIgnoredHeaders(t *testing.T) {
+	cases := map[string]struct {
+		Header        string
+		ExpectIgnored bool
```

Comment:
Agree, the initial `ExcludeList` rule within v4 is also a little confusing since it is valid when inner rule is invalid. I have changed the `ExpectIgnored` to be true if the header is ignored.